src/Domain/SegmentId.hpp

```diff
@@ -0,0 +1,115 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+/// \file
+/// Defines class SegmentId.
+
+#pragma once
+
+#include <cstdint>
+#include <iosfwd>
+#include <pup.h>
+
+#include "Domain/Side.hpp"
+#include "ErrorHandling/Assert.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+
+/// \ingroup ComputationalDomain
+/// A SegmentId labels a segment of the interval [-1,1].
+/// A SegmentId is used to (partially) identify an Element.
+/// The segment is the interval [-1 + 2*(N/D), -1 + 2*(N+1)/D]
+/// where N = index_ and D = 2 ^ refinement_level_.
+class SegmentId {
+ public:
+  /// Default constructor needed for Charm++ serialization.
+  SegmentId();
```

Comment:
make this `= default` and move the default values set in the constructor into the class definition